src/Valleysoft.DockerCredsProvider/CredsProvider.cs

```diff
@@ -5,25 +5,34 @@ namespace Valleysoft.DockerCredsProvider;
 public static class CredsProvider
 {
     public static Task<DockerCredentials> GetCredentialsAsync(string registry) =>
-        GetCredentialsAsync(registry, new FileSystem(), new ProcessService());
+        GetCredentialsAsync(registry, new FileSystem(), new ProcessService(), new Environment());
 
-    internal static async Task<DockerCredentials> GetCredentialsAsync(string registry, IFileSystem fileSystem, IProcessService processService)
+    internal static async Task<DockerCredentials> GetCredentialsAsync(string registry, IFileSystem fileSystem, IProcessService processService, IEnvironment environment)
     {
         if (registry is null)
         {
             throw new ArgumentNullException(nameof(registry));
         }
 
-        ICredStore credStore = await GetCredStoreAsync(registry, fileSystem, processService);
+        ICredStore credStore = await GetCredStoreAsync(registry, fileSystem, processService, environment);
         return await credStore.GetCredentialsAsync(registry);
     }
 
-    private static async Task<ICredStore> GetCredStoreAsync(string registry, IFileSystem fileSystem, IProcessService processService)
+    private static string GetConfigFilePath(IEnvironment env) {
+        if (env.GetEnvironmentVariable("DOCKER_CONFIG") is string configDirectory 
+            && !System.String.IsNullOrEmpty(configDirectory)) {
+                return Path.Combine(configDirectory, "config.json");
+            } else {
+                return Path.Combine(
+                    env.GetUserProfilePath(),
```

Comment:
I'd prefer the IEnvironment wrapper to more closely mimic the `System.Environment` class by having a `GetFolderPath` method instead.

Reply:
I can do that, but I was going for least-power principle here - we don't need all of the power of the more general method so it's easier to understand what the interface is aiming to do.